app/models/marking_weight.rb

```diff
@@ -1,3 +1,16 @@
 class MarkingWeight < ActiveRecord::Base
   belongs_to :marking_scheme
+
+  def get_gradable_item
+    if self.is_assignment
+      gradable_item = Assignment.where(id: gradable_item_id).first
+    else
+      gradable_item = GradeEntryForm.where(id: gradable_item_id).first
+    end
+    return gradable_item
```

Comment:
This `return` is superfluous. Ruby automatically returns the last expression evaluated in a method body. So in fact you don't need to assign to `gradable_item`; you can just evaluate either `Assignment....` or `GradeEntryForm...` and that will be returned.